cmd/swapd/main.go

```diff
@@ -298,30 +291,42 @@ func newEmptyDaemon(ctx context.Context, cancel context.CancelFunc) *daemon {
 }
 
 func (d *daemon) stop() error {
-	if d.database != nil {
-		err := d.database.Close()
```

Comment:
I didn't like that the database was getting closed while services that modify the database are still running, so I reversed the shutdown order.  I also didn't want the failure to shut down one service to prevent us from shutting down the other services.  I'm not thrilled with the result, but it was the best I could come up with while restraining the refactor to just this function.